aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -247,4 +251,55 @@ static inline std::tuple<std::vector<int64_t>,
   return std::make_tuple(q_sizes, q_strides, n_columns_q);
 }
 
+// Function to generate empty tensors of required size, strides and dtype for the SVD operation
+static inline std::tuple<Tensor, Tensor, Tensor> _create_U_S_VT(const Tensor& input, bool some, bool compute_uv) {
+  auto sizes = input.sizes().vec();
+  int64_t m = input.size(-2), n = input.size(-1);
+
+  sizes[input.dim() - 1] = (compute_uv && some) ? std::min(m, n) : m;
+  auto strides = at::detail::defaultStrides(sizes);
+  // U should be a column-major or a batch of column-major matrices
+  // ... x m x ucol will have strides: ...., ucol, 1
+  // We require: ...., 1, m
+  strides[input.dim() - 1] = m;
+  strides[input.dim() - 2] = 1;
+
+  Tensor U_empty;
+  if (!input.is_cuda()) {
+    U_empty = at::empty_strided(sizes, strides, input.options());
+  } else {
+    U_empty = at::empty_strided(sizes, strides, input.options().device(at::kCPU));
```

Comment:
why does these need to be on CPU tensors? Do they get moved back to cuda later?

Reply:
Yes, they are moved back in _svd_helper_cuda.

The MAGMA routine accepts matrices on CPU only, since it is a hybrid algorithm.

Reply:
ah OK, that makes sense. Maybe we can add that explanation in a code comment somewhere -- it's a bit confusing, at least to me, to see the `to(kCPU)` in a cuda op.

Reply:
Yep, I will add a comment.